spec/Section 5 -- Validation.md

```diff
@@ -1855,6 +1855,26 @@ variable.
 
 ### All Variable Usages Are Allowed
 
+As GraphQL evolves and issues are spotted and fixed, the specification aims to
+maintain compatibility with existing GraphQL documents. In rare cases, this
+requires two versions of an algorithm, a Legacy Algorithm which supports clients
+with both old and new GraphQL documents, and a Strict Algorithm which only
+supports new GraphQL documents. It is recommended that you implement the Strict
+Algorithm unless you have existing GraphQL documents that you need to support
+which would become invalid under the Strict Algorithm, in which case you should
+implement the Legacy Algorithm.
```

Comment:
"In which case you must implement the Legacy Algorithm until the existing documents are migrated to satisfy the strict algorithm"